go/api/response_handlers.go

```diff
@@ -0,0 +1,55 @@
+// Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+
+package api
+
+// #cgo LDFLAGS: -L../target/release -lglide_rs
+// #include "../lib.h"
+import "C"
+
+import (
+	"fmt"
+	"go/types"
+	"reflect"
+	"unsafe"
+)
+
+type redisResponse interface {
+	interface{} | string | types.Nil
+}
+
+func handleRedisResponse[T redisResponse](t reflect.Type, response interface{}) (T, error) {
+	if reflect.TypeOf(response) == t {
+		return reflect.ValueOf(response).Interface().(T), nil
+	}
+
+	var responseTypeName string
+	if response == nil {
+		responseTypeName = "nil"
+	} else {
+		responseTypeName = reflect.TypeOf(response).Name()
+	}
+
+	return reflect.Zero(t).Interface().(T), &RequestError{
+		fmt.Sprintf("Unexpected return type from Redis: got %s, expected %s", responseTypeName, t),
+	}
+}
+
+func convertCharArrayToString(arr *C.char, length C.long) string {
```

Comment:
Server (and Glide code too) can return a `BulkString` value, which is not convertable to a human-readable string. In such cases it should be interpreted as `byte[]`. Keep in mind that this applies to complex return types too. For example, `xread` returns `Map<GlideString, Map<GlideString, GlideString[][]>>` where `GlideString = Union[String, byte[]]`.
Command arguments could be `byte[]` too.
As earlier you start implementing this, as easier that will be. Consult with @eifrah-aws for more details.

Reply:
A simple example of commands with binary strings - a pair of `dump` and `restore`. You can play with it using `CustomCommand` API.

Reply:
GoString don't have any restrictions on the valid encoding of the string. Any ByteString can be represented as a valid GoString. Added an integration test to validate such scenarios.